src/Components/Game.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState } from "react";
+import Board from "./Board";
+import { GameProps, isNotNull } from "../Types/TypeProps";
+import GameInfo from './GameInfo';
+/**
+ * Handle Game logic
+ * @param props 
+ * @returns 
+ */
+const Game = (props:GameProps) => {
+  /**
+   * Game's state declarations
+   */
+  const [history, setHistory] = useState(props.historyVal);
+  const [stepNumber, setStepNumber] = useState(props.stepNumberVal);
+  const [xIsNext, setXisNext] = useState(props.xIsNextVal);
+  const [counter, setCounter] = useState(history.length);
+
+  /**
+   * Handle each board's cell action when clicked
+   * @param {*} i
+   */
+  const handleClick = (i: number) => {
+    const gameHistory = history.slice(0, stepNumber + 1);
+    const current = gameHistory[gameHistory.length - 1];
+    const squares = current.squares.slice();
+    
+
+    // Prevent further cell clicks after winning
+    if (calculateWinner(squares) || squares[i]) return;
+
+    // Determine next player
+    squares[i] = xIsNext ? "X" : "O";
+
+    // Update states
+    setHistory(gameHistory.concat([{ squares: squares }]));
```

Comment:
```suggestion
    setHistory([...gameHistory, [{ squares: squares }]]);
```

I would use the spread syntax here instead